### DISCUSSION TITLE
GTK Header bar on Godot builds for Linux.

### DISCUSSION THREAD

**Discussion by vvvvvvitor** [godotengine/godot-proposals#6370] [Platforms]
I made a mockup of what the header bar for GTK applications could look in Godot, I'm no graphic designer, it's just an idea I've had for a while. Godot's UI looks really different when compared to GTK's, this could help bring some consistent and also free some screen space. I've read that Wayland is adding support for [client-side window decorations](https://github.com/electron/electron/pull/29618) so this may be achievable.

![empty-page](https://user-images.githubusercontent.com/119557151/221737062-c7fc1017-c5e1-409e-a2e6-697599c3b64e.svg)

**Comment by zinnschlag** [comment]
If this feature is implemented then please make it optional. There are still plenty of Linux desktops out there that use conventional title bars.

**Reply by vvvvvvitor** [reply]
Yes, it would be optional, just like OSX's titlebar.

**Comment by bruvzg** [comment]
For the reference: Godot 4 already have similarly looking extend-to-title option on macOS, it might eventually have it on Linux as well. Godot do not use GTK, and adding it as a dependency just for the title bar is not acceptable, so it will look differently anyway.

**Reply by vvvvvvitor** [reply]
That's why I suggested doing it with Wayland's client-side decorations
